izumi-reflect/izumi-reflect/src/main/scala-2/izumi/reflect/TagMacro.scala

```diff
@@ -453,10 +453,43 @@ class TagMacro(val c: blackbox.Context) {
     tpeSymbol
   }
 
+  protected[this] def mkTypeParameter(owner: Symbol, tpe: Type): Symbol = {
+    import internal.reificationSupport._
+    import internal.{polyType, typeBounds}
+
+    val tpeSymbol = newNestedSymbol(owner, freshTypeName(""), NoPosition, Flag.PARAM | Flag.DEFERRED, isClass = false)
+
+    val tpeTpe = tpe match {
+      case PolyType(params, _) =>
+        val newParams = params.map(p => mkTypeParameter(tpeSymbol, p.typeSignature))
```

Comment:
I wonder if instead of looping here it wouldn't be more elegant to add TypeBounds to `Kind` data type and extract it in `kindOf`? Then you could in `protected[this] def mkTypeParameter(owner: Symbol, kind: Kind)` merely read the type bounds from `Kind` structure rather than perform pattern match on `Type` inside `mkTypeParameter`